### ISSUE TITLE
Array of actions 

### ISSUE THREAD

**Issue by Zaharah** [tensorforce/tensorforce#808]
First of all, thanks for well-documented and neat library.  
I am using tensorforce with a custom environment to solve maintenance planning problem. My action space looks as follows: 
`def actions(self):
    return dict(type='int', num_values=3, shape=(16,))`

In each timestep, the agent would return output like so e.g. `[1 0 0 1 1 0 2 0 0 1 2 1 0 2 2 1]`. 
I am using standard configuration for agent as explained [here](https://tensorforce.readthedocs.io/en/latest/basics/getting-started.html). 

After some training and also in evaluation, the agent return exact same array of actions for all (100) timesteps instead of outputting optimal actions. I could not find any examples where agent takes an array of actions instead of single discrete action.

Could you please tell me what could be the problem? Why the agent returns exact same array of actions for all timesteps? What I am possibly doing wrong?  Thanks in advance for your help.

**Comment by AlexKuhnle** [comment]
Hi @Zaharah,

The reason for this is usually that the environment (or potentially agent) is configured in such a way that it doesn't learn anything useful and instead collapses to always choose the same action (e.g. do nothing in the case of maintenance, where most of the times this is the "best" action). Could you provide more detail about the meaning of actions, the state space, and how your reward works?

**Comment by Zaharah** [comment]
Thanks for the prompt response @AlexKuhnle. 
I am aiming to develop optimal maintenance plan for 16 water pipes.  For each pipe, there can be three possible actions as 0: do nothing, 1: Maintain and 2: Replace. The shape of state space is (16,6) with attributes like probability of failure, reliability level, age, material, diameter, failure rate.  The reward function looks like this: 

`costs[pipe] = np.where(action == 0 and pf_pipe >= 0.9, -1,   
                                  np.where(action == 2, -0.8,
                                  np.where(action == 1 and pf_pipe > 0.5, -0.5,
                                  np.where(action == 1 and pf_pipe<=0.5, -1, 0) )))`

The probability of failure and reliability level are inversely proportional to each other.  These values change depending on the chosen action.  
The agent configuration is as follows;

`agent = Agent.create(
agent='tensorforce', environment=environment, update=64,
optimizer=dict(optimizer='adam', learning_rate=1e-3),
objective='policy_gradient', reward_estimation=dict(horizon = 1)
)`

Many thanks for your help.

**Comment by AlexKuhnle** [comment]
As far as I can tell, that means that `action = 1` and `action = 2` basically always result in a negative reward (shouldn't that sometimes be positive?), whereas `action = 0` only does if `pf_pipe >= 0.9`. If this latter case doesn't happen very often, particularly with some randomness in the actions, then the agent will get a very strong signal that `action = 0` is the most optimal thing to do. Remember that the agent initially acts randomly and needs to learn to act consistently first, which may well mean that it (almost) never even encounters the case of `pf_pipe >= 0.9` (assuming maintain and replace both reset this?) at the beginning of training, and then later on has already overfitted to the fact that action 1 and 2 are bad.

**Comment by Zaharah** [comment]
Perhaps you are right about reward function. I will modify it to see if it works. 
One last question, are you aware of any example(s) of tensorforce for learning optimal array of actions instead of single discrete action?

**Comment by AlexKuhnle** [comment]
I couldn't point you to an open implementation or similar, at least not of the top of my head, but I have come across such applications via questions from users, yes. Plus, multiple action values is common for many of the standard continuous benchmarks.

**Comment by AlexKuhnle** [comment]
Assuming this can be closed then.
